File: packages/extract/test/fixtures/polyglot/c/job.c
#include <stdlib.h>
#include "job.h"

typedef enum { IDLE, RUNNING, DONE, FAILED, RETRYING } job_state;

static job_state state = IDLE;

void job_start(void) {
    if (state != IDLE) return;
    state = RUNNING;
}

void job_finish(void) {
    if (state == RUNNING) state = DONE;
}

void job_fail(void) {
    state = FAILED;
}

const char *job_describe(void) {
    switch (state) {
    case IDLE: return "waiting";
    case RUNNING: return "busy";
    case DONE: return "finished";
    }
    return "";
}

int job_process(int n) {
    char *buffer = malloc(n);
    if (buffer == NULL) return -1;
    int ok = n > 0;
    if (n > 100) return -2;
    free(buffer);
    return ok;
}

File: packages/extract/test/fixtures/polyglot/go/job.go
package job

import "os"

type State int

const (
	Idle State = iota
	Running
	Done
	Failed
	Retrying
)

type Job struct {
	state State
}

func NewJob() *Job {
	return &Job{state: Idle}
}

func (j *Job) Start() {
	if j.state != Idle {
		return
	}
	j.state = Running
}

func (j *Job) Finish() {
	if j.state == Running {
		j.state = Done
	}
}

func (j *Job) Fail() {
	j.state = Failed
}

func (j *Job) Describe() string {
	switch j.state {
	case Idle:
		return "waiting"
	case Running:
		return "busy"
	case Done:
		return "finished"
	}
	return ""
}

func FirstByte(path string) (byte, error) {
	f, err := os.Open(path)
	if err != nil {
		return 0, err
	}
	buf := make([]byte, 1)
	if _, err = f.Read(buf); err != nil {
		return 0, err
	}
	f.Close()
	return buf[0], err
}
